Can not build on raspberry pi - /usr/bin/ld: libopenrct2.a(Path.cpp.o): in function `std::filesystem::__cxx11::path::path

**OS:**  Raspbian (Debian 10 armv7 armhf)
**Version:** defaut branch

<!-- Explanation of the issue -->



- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. git clone https://github.com/OpenRCT2/OpenRCT2
2. mkdir build
3. cd build
4. cmake ../
5. make

error log:
https://pastebin.com/NibRkFm7

> Same issue I reported in #10352. I do have a hacky workaround in that issue to manually link in the required library.


Thanks for the quick fix! However, this is what I got after the quick fix:


`-- Performing Test CXX_WNO_CLOBBERED - Success
-- Performing Test CXX_WARN_REDUNDANT_DECLS
-- Performing Test CXX_WARN_REDUNDANT_DECLS - Success
CMake Error at CMakeLists.txt:219 (target_link_libraries):
  Cannot specify link libraries for target "libopenrct2" which is not built
  by this project.


-- Configuring incomplete, errors occurred!
See also "/home/pi/game/OpenRCT2/build/CMakeFiles/CMakeOutput.log".
See also "/home/pi/game/OpenRCT2/build/CMakeFiles/CMakeError.log".`

--- a/distribution/changelog.txt
+++ b/distribution/changelog.txt
@@ -11,6 +11,7 @@
 - Fix: [#23486] Object selection minimum requirements can be bypassed with close window hotkey.
 - Fix: [#23743] Parks with guest goals over 32767 do not appear in the scenario list.
 - Fix: [#23844] Sound effects keep playing when loading another save.
+- Fix: [#23881] Compiling on Raspbian/arm-linux-gnueabihf fails.
 - Fix: [#23891] Inverted Hairpin Coaster track can draw over things above it (original bug).
 - Fix: [#23892] Gentle banked Wooden Roller Coaster track glitches as trains pass (original bug).
 - Fix: [#23897] Reverse Freefall Coaster slope up to vertical track piece does not draw a vertical tunnel.

--- a/src/openrct2/CMakeLists.txt
+++ b/src/openrct2/CMakeLists.txt
@@ -288,3 +288,8 @@ endif ()
 if (UNIX)
     add_definitions(-D_FILE_OFFSET_BITS=64)
 endif ()
+
+if (CMAKE_LIBRARY_ARCHITECTURE MATCHES "arm-linux-gnueabihf")
+    message(STATUS "Linking to armhf libs; adding libatomic")
+    target_link_libraries(${PROJECT_NAME} atomic)
+endif()

